README.md

```diff
@@ -414,6 +414,22 @@ export default class UserProfile extends Component {
 
 This also holds true for all other macros of the ember-decorators addon.
 
+#### Typing uninitialized properties
+
+For folks using babel@7 you may encounter problems when providing type definitions such as the following when working with Ember below `3.6`.
+
+```ts
+import Component from '@ember/component';
+
+export default class UserProfile extends Component {
+  username?: string;
+}
+```
+
+When invoked via a template `{{user-profile username='example123'}}`, you would expect that `username` would have the value of `example123`, however prior to the native class feature released in Ember `3.6`, this will result in `username` being undefined.  
```

Comment:
We should consider whether it's preferable to just include `ember-native-class-polyfill` as either a dependency or something we check for in this addon's `included()` hook. Looks like the polyfill supports _at least_ Ember 3.4 (https://github.com/pzuraq/ember-native-class-polyfill/issues/11) which satisfies our framework version support policy.

The root cause of the problem this documentation is trying to help users avoid, is that babel 7 auto-initializes uninitialized properties to `void 0` (possibly for perf reasons, so runtimes see instances as having a consistent shape), which interferes with the way we expect `Ember.Component` (and `Glimmer.Component`) to work.

Reply:
Initializing the field as `void 0` / `undefined` has the side-effect of being good for perf, but Babel primarily does this to be spec-compliant.

https://github.com/babel/babel/issues/9105 and  https://github.com/babel/babel/pull/9141 go into detail about this.